ledger/applications.go

```diff
@@ -85,6 +87,15 @@ func (al *logicLedger) MinBalance(addr basics.Address, proto *config.ConsensusPa
 	return record.MinBalance(proto), nil
 }
 
+func (al *logicLedger) Authorizer(addr basics.Address) basics.Address {
+	record, err := al.cow.Get(addr, false) // pending rewards unneeded
+	if err != nil || record.AuthAddr.IsZero() {
```

Comment:
If `err != nil`, does that mean no account exists for `addr`, so it's ok to interpret it as having no AuthAddr?

Reply:
Correct.  An account needs to exist to hold a rekey address to be rekeyed.  When it doesn't exist, it's controlled the usual way. That often means it can't do anything anyway, but it could be a logicsig or, now, an app address that simply hasn't been funded yet. It *could* still "sign" a txn.

Reply:
I'm going to spend a little time digging into this.  I share your concern. At the very least, perhaps we should return an error that explicitly communicates "NoAccount".  Then this is safe.

As I started thinking about how we handle this for top-level transactions, I realized we probably don't - if the account doesn't exist, who cares what its AuthAddr is? It can't participate in transactions, because it has nothing and can't even pay fees!

Except, no.  Nowadays, a properly authorized transaction should be able to do a clawback transaction, potentially without paying any fees, or having any algos or assets, because of fee pooling.  I have been told that this doesn't work and I've been meaning to investigate.  This is probably why - I bet we throw away a transaction that has this error.

Reply:
Widened interface to avoid ambiguity.